src/movies.js

```diff
@@ -1,16 +1,113 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies){
+
+const directorF =  movies.map(function(movie){
+        return movie.director
+    })
+
+return directorF
+}
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies){
+    let drama;
+    let spiel = movies.filter((movie) => {
+    for (let i=0; i < movie.genre.length; i++){
+       drama = movie.genre[i] == "Drama" && movie.director == "Steven Spielberg";
+       if(drama){break;}
+    } 
+    return drama
+}) 
+return spiel.length;
+};
+
+function howManyMovies(movies){
+    let Drama = movies.filter(function(ele){
+      return ele.genre.includes("drama")
+    })
+    return Drama;
+  }
+
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movies){
+  
+    let sum = movies.reduce(function(acc,value){
+      return acc + value.rate
+    },0)
+    return sum/sum.length;
+  }
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(movies){
+    let s = movies.filter(function(elw){
```

Comment:
As best practices, avoid these kind of names for your variables. Use names that reflect the expected value that will be stored inside. This is very important for other developers checking on your code and even for you when you go back to your code.